phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java

```diff
@@ -3678,6 +3679,34 @@ protected PhoenixConnection upgradeSystemCatalogIfRequired(PhoenixConnection met
                 MIN_SYSTEM_TABLE_TIMESTAMP_4_16_0,
                 PhoenixDatabaseMetaData.PHOENIX_TTL_HWM + " "
                         + PInteger.INSTANCE.getSqlTypeName());
+
+            boolean isNamespaceMapping =
+                    SchemaUtil.isNamespaceMappingEnabled(null,  getConfiguration());
+            String tableName = PhoenixDatabaseMetaData.SYSTEM_CATALOG_NAME;
+            if (isNamespaceMapping) {
+                tableName = tableName.replace(
+                        QueryConstants.NAME_SEPARATOR,
+                        QueryConstants.NAMESPACE_SEPARATOR);
+            }
+
+            byte[] rowKey = SchemaUtil.getColumnKey(null,
+                    QueryConstants.SYSTEM_SCHEMA_NAME,
+                    SYSTEM_CATALOG_TABLE, VIEW_INDEX_ID,
+                    PhoenixDatabaseMetaData.TABLE_FAMILY);
+            if (UpgradeUtil.isUpdateViewIndexIdColumnDataTypeFromShortToLongNeeded
+                    (metaConnection, rowKey, tableName.getBytes())) {
+                try {
+                    LOGGER.info("Updating VIEW_INDEX_ID data type to BIGINT.");
+                    UpgradeUtil.updateViewIndexIdColumnDataTypeFromShortToLong(
+                            metaConnection, rowKey, tableName.getBytes());
+                    LOGGER.info("Updated VIEW_INDEX_ID data type from SMALLINT TO BIGINT.");
+                } catch (Exception e) {
+                    LOGGER.error("Updating VIEW_INDEX_ID data type failed.");
```

Comment:
Change to : `LOGGER.error(" ... ", e)` so we see the actual error too

Reply:
Or maybe that's not required since you're already catching the exception and logging it inside the updateViewIndexIdColumnDataTypeFromShortToLong() method. Can this method still throw then?